lib/save_decorator.rb

```diff
@@ -0,0 +1,171 @@
+        first_name: instance.first_name,
+        last_name: instance.last_name,
+        item_ids: instance.items.map(&:id)
+      }
+    end
+  end
+
+  def save_json
+    File.write('./data/authors.json', to_json)
+  end
+end
+
+class SaveLabelDecorator < SaveDecorator
+  def hash_generator
+    @instances.each do |instance|
+      @hash[instance] = {
+        id: instance.id,
+        title: instance.title,
+        color: instance.color,
+        item_ids: instance.items.map(&:id)
+      }
+    end
+  end
+
+  def save_json
+    File.write('./data/labels.json', to_json)
+  end
+end
+
+class SaveGenreDecorator < SaveDecorator
+  def hash_generator
+    @instances.each do |instance|
+      @hash[instance] = {
+        id: instance.id,
+        name: instance.name,
+        item_ids: instance.items.map(&:id)
+      }
+    end
+  end
+
+  def save_json
+    File.write('./data/genres.json', to_json)
+  end
+end
+
+class SaveItemDecorator < SaveDecorator
+  def hash_generator
+    @instances.each do |instance|
+      @hash[instance] = {
+        id: instance.id,
+        item_class: instance.class.to_s,
+        genre: genre_hash(instance),
+        author: author_hash(instance),
+        label: label_hash(instance),
+        publish_date: instance.publish_date,
+        archived: instance.archived,
+        class_properties: class_properties_generator(instance)
+      }
+    end
+  end
+
+  def save_json
+    File.write('./data/items.json', to_json)
+  end
+end
+
+def class_properties_generator(instance)
+  case instance.class.to_s
+  when 'Book'
+    book_class(instance)
+  when 'Game'
+    game_class(instance)
+  when 'MusicAlbum'
+    music_class(instance)
+  else
+    raise NotImplementedError, 'Instance class not implemented yet'
+  end
+end
+
+def book_class(instance)
+  {
+    publisher: instance.publisher,
+    cover_state: instance.cover_state
+  }
+end
+
+def game_class(instance)
+  {
+    multiplayer: instance.multiplayer,
+    last_played_date: instance.last_played_date
+  }
+end
+
+def music_class(instance)
+  {
+    on_spotify: instance.on_spotify
+  }
+end
+
+def genre_hash(instance)
+  {
+    genre_id: instance.genre.id,
+    genre_name: instance.genre.name
+  }
+end
+
+def author_hash(instance)
+  {
+    author_id: instance.author.id,
+    author_first_name: instance.author.first_name,
+    author_last_name: instance.author.last_name
+  }
+end
+
+def label_hash(instance)
+  {
+    label_id: instance.label.id,
+    label_title: instance.label.title,
+    label_color: instance.label.color
+  }
+end
+
+# class SaveGameDecorator < SaveItemDecorator
+#   def hash_generator
+#     @instances.each do |instance|
+#       @hash[instance] = {
+#         id: instance.id,
+#         multiplayer: instance.multiplayer,
+#         last_played_date: instance.last_played_date,
+#         items: instance.items.map(&:id)
+#       }
+#     end
+#   end
+
+#   def save_json
+#     File.write('./data/authors.json', to_json)
+#   end
+# end
```

Comment:
- [ ] Kindly remove all comments in your code, this is to ensure best practices and clean code